libs/npress/models/PagesModel.php

```diff
@@ -235,6 +235,101 @@ public static function cacheAllPages()
     }
   }
 
+  public static function addPagesHistory(
+    $id_page,
+    $lang,
+    $name,
+    $seoname,
+    $heading,
+    $text
+  ) {
+    dibi::query('INSERT INTO pages_history', array(
+      'id_page' => $id_page,
+      'lang' => $lang,
+      'id_parent' => 0,
+      'name' => $name,
+      'seoname' => $seoname,
+      'heading' => $heading,
+      'text' => $text,
+      'updated_at%SQL' => 'NOW()'
+    ));
+  }
+  public static function rollBackPage(
+    $id_page,
+    $lang,
+    $name,
+    $seoname,
+    $heading,
+    $text
+  ){
+    dibi::query('INSERT INTO pages', array(
+      'id_page' => $id_page,
+      'lang' => $lang,
+      'id_parent' => 0,
+      'name' => $name,
+      'seoname' => $seoname,
+      'heading' => $heading,
+      'text' => $text,
+      'ord' => 0,
+      'published' => 0,
+      'deleted' => 0
+    ), 'ON DUPLICATE KEY UPDATE %a', [ // tady už modifikátor %a uvést musíme
+      'lang' => $lang,
+      'id_parent' => 0,
+      'name' => $name,
+      'seoname' => $seoname,
+      'heading' => $heading,
+      'text' => $text,
+      'ord' => 0,
+      'published' => 0,
+      'deleted' => 0
+    ]);
+  }
+  public static function getAllPagesHistory()
+  {
+    $history = dibi::fetch("SELECT * FROM pages_history ");
+    return $history;
+  }
+
+  public static function getHistoryIdOnLink($id_page)
+  {
+    $id_history = dibi::fetch(
+      "SELECT * FROM pages_history WHERE id_page = ",
+      $id_page
+    );
+    return $id_history;
+  }
+
+  public static function getAllExceptLast($id_page)
+  {
+    $id_history = dibi::fetchAll(
+      "SELECT * FROM `pages_history` WHERE NOT updated_at  IN (SELECT max(updated_at)FROM pages_history) AND id_page = ",
+      $id_page
```

Comment:
1. verze chceš vypsat asi od nejnovější. Čili přidej `ORDER BY updated_at DESC`. Teď to vypadá takhle, což je matoucí:
![image](https://user-images.githubusercontent.com/385047/117033747-f9f42900-ad02-11eb-80c7-bd69803c9840.png)


2. proč si nevybereš i tu poslední? Ta logika by celá byla daleko jednodušší. V tom cyklu kde to vypisuješ prostě i prvního řádku (`$iteration == 0`) nevypíšeš tlačítko ale "toto je aktuální verze". Co se týče linku pro "návrat zpět k editaci", tam můžeš využít `$current->id_page`